### PR TITLE
Convert GH actions into Kubernetes CronJobs

### PR DESCRIPTION
**Pull request by mac-chaffee**
k8s-diagram

This is a proof of concept for running haloperidol inside of a Kubernetes cluster. This would allow you to test behaviors of the [qdrant-operator](https://github.com/qdrant/qdrant-operator) inside of our staging Kubernetes cluster in a dedicated namespace.

The benefits of this approach are:
* You can have as many long-running multi-node qdrant clusters as you want, just make more QdrantCluster yaml files like this one
* You can directly connect to the qdrant cluster using its internal hostname, no need to manage authentication or try to connect from a github action to AWS
* No additional infrastructure needed

### TODO:
- [x] Set up flux to auto-apply these manifests to staging ([done](https://github.com/qdrant/qdrant-cloud/pull/1131))
- [x] Finish converting all the cronjobs
- [x] Finish converting all the scripts to no longer need docker
- [x] Test
- [ ] Configure alerts for when the cronjobs fail. We have existing monitoring infrastructure that can trigger alerts when cronjobs fail, but it would need to be configured to send these alerts to a different slack channel.
- [ ] Verify permissions on this repo. Might should make it private. Anyone with push access can deploy stuff in our staging cluster, which isn't horrible but we should just remove that risk.

### CODE REVIEW

File: kubernetes/cronjobs.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            app: qdrant
+            cluster-id: haloperidol
+            customer-id: haloperidol
+        spec:
+          containers:
+          - name: haloperidol
+            image: qdrant/bfb:latest
+            env:
+            - name: QDRANT_URL
+              value: http://qdrant-haloperidol.haloperidol.svc.cluster.local:6333
```
Comment (mrIncompetent) [inline kubernetes/cronjobs.yaml:133]: fqdn :tada:

**Review by mrIncompetent — APPROVED** [review]

**Comment by mac-chaffee** [comment]
@KShivendu I believe I've finished converting all the scripts now!

Do you know if the "bfb-upload" script is supposed to run multiple times? Or just once to load the initial data?

I see the script is called from Github Actions on a daily schedule, but when I run my converted version of that script twice, it fails the second time due to the collection already existing.

If `bfb-upload` is only meant to run once, I can convert the CronJob into a normal Job that just runs once at install time.

**Comment by KShivendu** [comment]
Afaik Bfb-upload is supposed to run all the time. The cron just keeps on updating it and checking that it's alive. 

We want to test if there's any degradation when running Qdrant for a long time.

**Comment by mac-chaffee** [comment]
Do the existing crons ever delete the data? Since bfb throws errors if the collection already exists, not sure how the existing crons would work without deleting the collection at some point, but I don't see that in this repo.

I can convert bfb-upload into a deployment, but I think I'd beed bfb to be updated to not throw errors if the collection exists.

**Comment by mac-chaffee** [comment]
Maybe I was just mistaken. It seems like `--create-if-missing` is working correctly after all.

I think this is fully ready now! Both bfb-upload and bfb-search will now run forever against the cluster, with the cronjobs triggering the random node failures and the snapshots every so often.

**Comment by KShivendu** [comment]
Great work @mac-chaffee 👏

I'll continue my part as soon as I'm back from vacations. Please avoid merging this before that as it will break the currently running bfb setup with github actions.

**Comment by mac-chaffee** [comment]
Actually this code doesn't share anything with the existing code, these are 100% new files (there was a point where I did edit the old files, but I undid that commit). We'd be safe to merge I think.

Once https://github.com/qdrant/qdrant-cloud/pull/1131 is merged too, it'll get deployed to staging

**Comment by mac-chaffee** [comment]
About Grafana: not sure how much news has gotten to you, but we have a new set of Grafana instances ("new monitoring stack").

The staging cluster is wired up to this grafana instance: https://grafana-monitoring.staging-cloud.qdrant.io/

Must connect to Tailscale to access. I can get you access if you don't have it yet. You can work on the dashboard from within Grafana itself, but to get the changes to persist, we'll have to save the dashboard here: https://github.com/qdrant/qdrant-cloud/tree/main/environments/staging/apps/kube-prometheus-stack/dashboards/qdrant-custom-dashboards

**Review by KShivendu — APPROVED** [review]
Nice it's good to go then. Thanks!

**Pull request merged**